File: Sources/SilknetDS/Components/SilknetHelperText.swift
// SilknetHelperText — small caption shown standalone or by Input/TextArea.
// Default state = plain text only. Semantic states (success/info/warning/
// error) prepend a 16pt SF Symbol and switch to the matching text color.
// Mirrors React HelperText 1:1 (including the Figma-source typo:
// `--text-warrning` → `Color.silknet.textWarrning`).

import SwiftUI

public enum SilknetHelperTextState: Sendable {
    case `default`
    case success
    case info
    case warning
    case error
}

public struct SilknetHelperText: View {
    private let text: String
    private let state: SilknetHelperTextState

    public init(_ text: String, state: SilknetHelperTextState = .default) {
        self.text = text
        self.state = state
    }

    public var body: some View {
        HStack(spacing: .silknet.digitsSpacing2) {
            if let symbol = iconName {
                Image(systemName: symbol)
                    .resizable()
                    .scaledToFit()
                    .frame(width: 16, height: 16)
                    .foregroundColor(textColor)
            }
            Text(text)
                .font(.silknet.otherCaption)
                .foregroundColor(textColor)
                .kerning(0.25)
        }
    }

    private var iconName: String? {
        switch state {
        case .default: return nil
        case .success: return "checkmark.circle.fill"
        case .info:    return "info.circle.fill"
        case .warning: return "exclamationmark.triangle.fill"
        case .error:   return "xmark.octagon.fill"
        }
    }

    private var textColor: Color {
        switch state {
        case .default: return .silknet.textAdditional
        case .success: return .silknet.textSuccess
        case .info:    return .silknet.textInfo
        case .warning: return .silknet.textWarrning // Figma source spelling.
        case .error:   return .silknet.textError
        }
    }
}

#if DEBUG
#Preview("All states") {
    VStack(alignment: .leading, spacing: 12) {
        SilknetHelperText("Default helper text")
        SilknetHelperText("Looking good", state: .success)
        SilknetHelperText("Just so you know", state: .info)
        SilknetHelperText("Heads up", state: .warning)
        SilknetHelperText("Something went wrong", state: .error)
    }
    .padding(24)
    .background(Color.silknet.backgroundLayer)
}
#endif

File: android/library/src/main/kotlin/ge/silknet/ds/components/SilknetHelperText.kt
// SilknetHelperText — Compose mirror of iOS SilknetHelperText.
// Caption text; default state has no icon, semantic states prepend a 16dp
// Material icon and switch text color.

package ge.silknet.ds.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material.icons.filled.Error
import androidx.compose.material.icons.filled.Info
import androidx.compose.material.icons.filled.Warning
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import ge.silknet.ds.Silknet
import ge.silknet.ds.SilknetTheme

enum class SilknetHelperTextState { Default, Success, Info, Warning, Error }

@Composable
fun SilknetHelperText(
    text: String,
    modifier: Modifier = Modifier,
    state: SilknetHelperTextState = SilknetHelperTextState.Default,
) {
    val color = colorFor(state)
    val icon = iconFor(state)
    Row(
        modifier = modifier,
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(Silknet.dimens.digitsSpacing2),
    ) {
        if (icon != null) {
            Icon(
                imageVector = icon,
                contentDescription = null,
                tint = color,
                modifier = Modifier.size(16.dp),
            )
        }
        Text(
            text = text,
            color = color,
            style = Silknet.typography.otherCaption.copy(letterSpacing = 0.25.sp),
        )
    }
}

@Composable
private fun colorFor(state: SilknetHelperTextState): Color = when (state) {
    SilknetHelperTextState.Default -> Silknet.colors.textAdditional
    SilknetHelperTextState.Success -> Silknet.colors.textSuccess
    SilknetHelperTextState.Info    -> Silknet.colors.textInfo
    SilknetHelperTextState.Warning -> Silknet.colors.textWarrning // Figma source spelling.
    SilknetHelperTextState.Error   -> Silknet.colors.textError
}

private fun iconFor(state: SilknetHelperTextState): ImageVector? = when (state) {
    SilknetHelperTextState.Default -> null
    SilknetHelperTextState.Success -> Icons.Filled.CheckCircle
    SilknetHelperTextState.Info    -> Icons.Filled.Info
    SilknetHelperTextState.Warning -> Icons.Filled.Warning
    SilknetHelperTextState.Error   -> Icons.Filled.Error
}

@Preview(name = "All states")
@Composable
private fun PreviewHelperText() {
    SilknetTheme {
        Box(
            modifier = Modifier
                .background(Silknet.colors.backgroundLayer)
                .padding(24.dp)
        ) {
            Column(verticalArrangement = Arrangement.spacedBy(12.dp)) {
                SilknetHelperText("Default helper text")
                SilknetHelperText("Looking good", state = SilknetHelperTextState.Success)
                SilknetHelperText("Just so you know", state = SilknetHelperTextState.Info)
                SilknetHelperText("Heads up", state = SilknetHelperTextState.Warning)
                SilknetHelperText("Something went wrong", state = SilknetHelperTextState.Error)
            }
        }
    }
}
